rewrite-java/src/main/java/org/openrewrite/java/format/SpacesVisitor.java

```diff
@@ -436,12 +436,8 @@ public SpacesVisitor(SpacesStyle style) {
     @Override
     public J.NewArray visitNewArray(J.NewArray newArray, P p) {
         J.NewArray n = super.visitNewArray(newArray, p);
-        if (getCursor().getParent() != null && getCursor().getParent().getValue() instanceof J.Annotation) {
-            if (n.getPadding().getInitializer() != null) {
-                n = n.getPadding().withInitializer(
-                        spaceBefore(n.getPadding().getInitializer(), style.getBeforeLeftBrace().isAnnotationArrayInitializerLeftBrace())
-                );
-            }
+        if (getCursor().getParent() != null && getCursor().dropParentUntil(o -> o instanceof J).getValue() instanceof J.Annotation) {
```

Comment:
Can use `getCursor().firstEnclosing(J.class)`

Reply:
Apparently, firstEnclosing doesn't go to cursor.parent, so if cursor.value is a J, it returns that, so I have to use `getCursor().getParent().firstEnclosing(...)`